lightning/kv/sql2kv_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"bytes"
+	"fmt"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/types"
+	"go.uber.org/zap/zapcore"
+)
+
+type mockEncoder struct {
+	zapcore.ArrayEncoder
+	zapcore.ObjectEncoder
+	buf *bytes.Buffer
+}
+
+func (enc *mockEncoder) AppendObject(obj zapcore.ObjectMarshaler) error {
+	err := obj.MarshalLogObject(enc)
+	return err
+}
+
+func (enc *mockEncoder) AddInt(key string, value int) {
+	enc.buf.WriteString(fmt.Sprintf("[%v=%v]", key, value))
+}
+
+func (enc *mockEncoder) AddString(key string, value string) {
+	enc.buf.WriteString(fmt.Sprintf("[%v=%v]", key, value))
+}
+
+func (s *kvSuite) TestMarshal(c *C) {
+	nullDatum := types.Datum{}
+	nullDatum.SetNull()
+	minNotNull := types.Datum{}
+	minNotNull.SetMinNotNull()
+	datums := rowArrayMarshaler{types.NewStringDatum("1"), nullDatum, minNotNull, types.MaxValueDatum()}
+	encoder := &mockEncoder{buf: &bytes.Buffer{}}
```

Comment:
I suggest we just use a [`MapObjectEncoder`](https://godoc.org/go.uber.org/zap/zapcore#MapObjectEncoder) here so you don't even need `mockEncoder`.

```go
	encoder := zapcore.NewMapObjectEncoder()
	encoder.AddArray("test", &S{})
	c.Assert(encoder.Fields["test"], DeepEquals, []map[string]interface{} { ... })
```